app/components/dialogs/NewBoardDialog.tsx

```diff
@@ -41,9 +44,15 @@ interface NewBoardDialogProps {
   close: () => void;
 }
 
+const getAllBoards = (state: RootState) => {
```

Comment:
You would not need a selector. The SideBar has list of all boards already. You could pass them as a Component property.